src/__tests__/NavigationContainer.test.tsx

```diff
@@ -112,3 +121,132 @@ it('throws when nesting performTransaction', () => {
     "Only one transaction can be active at a time. Did you accidentally nest 'performTransaction'?"
   );
 });
+
+it('handle dispatching with ref', () => {
+  function CurrentParentRouter(options: DefaultRouterOptions) {
+    const CurrentMockRouter = MockRouter(options);
+    const ParentRouter: Router<NavigationState, MockActions> = {
+      ...CurrentMockRouter,
+
+      shouldActionPropagateToChildren() {
+        return true;
+      },
+    };
+    return ParentRouter;
+  }
+
+  function CurrentChildRouter(options: DefaultRouterOptions) {
+    const CurrentMockRouter = MockRouter(options);
+    const ChildRouter: Router<NavigationState, MockActions> = {
+      ...CurrentMockRouter,
+
+      shouldActionChangeFocus() {
+        return true;
+      },
+
+      getStateForAction(state, action) {
+        if (action.type === 'REVERSE') {
+          return {
+            ...state,
+            routes: state.routes.slice().reverse(),
+          };
+        }
+        return CurrentMockRouter.getStateForAction(state, action);
+      },
+    };
+    return ChildRouter;
+  }
+
+  const ChildNavigator = (props: any) => {
+    const { state, descriptors } = useNavigationBuilder(
+      CurrentChildRouter,
+      props
+    );
+
+    return descriptors[state.routes[state.index].key].render();
+  };
+
+  const ParentNavigator = (props: any) => {
+    const { state, descriptors } = useNavigationBuilder(
+      CurrentParentRouter,
+      props
+    );
+
+    return (
+      <React.Fragment>
+        {state.routes.map(route => descriptors[route.key].render())}
+      </React.Fragment>
+    );
+  };
+
+  const ref = React.createRef<NavigationContainerRef>();
+
+  const onStateChange = jest.fn();
+
+  const initialState = {
+    index: 1,
+    routes: [
+      {
+        key: 'baz',
+        name: 'baz',
+        state: {
+          index: 0,
+          key: '4',
+          routeNames: ['qux', 'lex'],
+          routes: [{ key: 'qux', name: 'qux' }, { key: 'lex', name: 'lex' }],
+        },
+      },
+      { key: 'bar', name: 'bar' },
+    ],
+  };
+
+  const element = (
+    <NavigationContainer
+      ref={ref}
+      initialState={initialState}
+      onStateChange={onStateChange}
+    >
+      <ParentNavigator>
+        <Screen name="foo">{() => null}</Screen>
+        <Screen name="bar">{() => null}</Screen>
+        <Screen name="baz">
+          {() => (
```

Comment:
Maybe we should add multiple navigators in screens to verify only the focused navigator handles this action (the focused screen could be the last one)

Reply:
yeah